dydx/dydxPresenters/dydxPresenters/_v4/Profile/Components/dydxProfileBalancesViewPresenter.swift

```diff
@@ -32,27 +32,36 @@ public class dydxProfileBalancesViewPresenter: HostedViewPresenter<dydxProfileBa
     public override func start() {
         super.start()
 
-        AbacusStateManager.shared.state.accountBalance(of: AbacusStateManager.shared.environment?.dydxTokenInfo?.denom)
-            .sink { [weak self] dydxAmount in
-                if let dydxAmount = dydxAmount {
-                    self?.viewModel?.walletAmount = dydxFormatter.shared.raw(number: Parser.standard.asNumber(dydxAmount), digits: 4)
+        let decimal = 4
+        let dydxTokenDenom = AbacusStateManager.shared.environment?.dydxTokenInfo?.denom
+        Publishers
+            .CombineLatest(
+                AbacusStateManager.shared.state.accountBalance(of: dydxTokenDenom),
+                AbacusStateManager.shared.state.stakingBalance(of: dydxTokenDenom)
+            )
+            .sink { [weak self] accountBalance, stakingBalance in
+                if let accountBalance = accountBalance {
+                    self?.viewModel?.walletAmount = dydxFormatter.shared.raw(number: Parser.standard.asNumber(accountBalance), digits: decimal)
                     self?.viewModel?.transferAction = {
                         Router.shared?.navigate(to: RoutingRequest(path: "/transfer", params: ["section": "transferOut"]), animated: true, completion: nil)
                     }
                 } else {
                     self?.viewModel?.walletAmount = "-"
                     self?.viewModel?.transferAction = nil
                 }
-            }
-            .store(in: &subscriptions)
 
-        AbacusStateManager.shared.state.stakingBalance(of: AbacusStateManager.shared.environment?.dydxTokenInfo?.denom ?? "")
-            .sink { [weak self] dydxAmount in
-                if let dydxAmount = dydxAmount {
-                    self?.viewModel?.stakedAmount = dydxFormatter.shared.raw(number: Parser.standard.asNumber(dydxAmount), digits: 4)
+                if let stakingBalance = stakingBalance {
+                    self?.viewModel?.stakedAmount = dydxFormatter.shared.raw(number: Parser.standard.asNumber(stakingBalance), digits: decimal)
                 } else {
                     self?.viewModel?.stakedAmount = "-"
                 }
+
+                let totalAmount = (accountBalance ?? 0.0) + (stakingBalance ?? 0.0)
+                if totalAmount != 0 {
+                    self?.viewModel?.totalAmount = dydxFormatter.shared.raw(number: Parser.standard.asNumber(totalAmount), digits: decimal)
+                } else {
+                    self?.viewModel?.totalAmount = "-"
```

Comment:
what if totalAmount is indeed 0?

Reply:
Good point.. Updated.

Reply:
i still think it might show "-" unintentionally here if, for example, `accountBalance = nil` and `stakingBalance = 1`

In this situation I think we'd want to show `1`